src/main/java/edu/hm/hafner/analysis/Issues.java

```diff
@@ -36,6 +36,118 @@
     private String path;
     private String moduleName;
 
+    /**
+     * Method to filter for issues.
+     *
+     * @param issueFilterInclude
+     *         include filter
+     * @param issueFilterExclude
+     *         exclude filter
+     *
+     * @return filtered issues
+     */
+    public Issues includedExcludedFilterForIssues(final IssueFilter issueFilterInclude, final IssueFilter issueFilterExclude) {
+        Issues tmpIssuesIncludedFiltered = getIncludedFilteredIssues(issueFilterInclude);
+        return getExcludedFilteredIssues(issueFilterExclude, tmpIssuesIncludedFiltered);
+    }
+
+    /**
+     * Method to filter all included issues.
+     *
+     * @param issueFilterInclude
+     *         issues to include
+     *
+     * @return filtered issues
+     */
+    private Issues getIncludedFilteredIssues(final IssueFilter issueFilterInclude) {
+        Issues tmpIssues = new Issues();
+        boolean issueFilteredInclude = false;
+
+        for (Issue issue : this) {
+            if (issueFilterInclude.getFileNames().contains(issue.getFileName())) {
+                issueFilteredInclude = true;
+                tmpIssues.add(issue);
+            }
+            if (issueFilterInclude.getPackageNames().contains(issue.getPackageName())) {
+                issueFilteredInclude = true;
+                tmpIssues.add(issue);
+            }
+            if (issueFilterInclude.getModuleNames().contains(issue.getModuleName())) {
+                issueFilteredInclude = true;
+                tmpIssues.add(issue);
+            }
+            if (issueFilterInclude.getCategories().contains(issue.getCategory())) {
+                issueFilteredInclude = true;
+                tmpIssues.add(issue);
+            }
+            if (issueFilterInclude.getTypes().contains(issue.getType())) {
+                issueFilteredInclude = true;
+                tmpIssues.add(issue);
+            }
+        }
+
+        /**
+         * There is no filter applied.
+         */
+        if (!issueFilteredInclude) {
```

Comment:
Für den Fall dass keine Issue die in issueFilterInclude angegebenen Include-Filter passiert, muss tmpIssues leer sein. Nur für den Fall, dass gar keine Include-Filter in issueFilterInclude definiert sind muss tmpIssues = this sein.